### PR TITLE
Fetch photo representing Trip and replace analog time picker with keyboard one

### PR DESCRIPTION
**Pull request by hieudan225**
1. Replace analog time picker with keyboard time picker. Since with keyboard time picker, user can submit an invalid time, I modified the `handleDataChange()` function of `ItemForm` to validate the date before submitting the change.

2. Fetch photos representing Trip using trip's`placeId` and save the photo Url in firebase along with other fields of `TripSetting`.

*Note: branch off UserPreferencing

### CODE REVIEW

**Comment by hieudan225** [comment]
image

File: step-capstone/src/components/TravelObjectForms/AddEvent.js [hunk 1]
```diff
@@ -1,146 +1,165 @@
...
+  Checkbox,
+  FormControlLabel,
+  Box,
+} from "@material-ui/core";
 import {
-    MuiPickersUtilsProvider,
-    DateTimePicker
-} from '@material-ui/pickers';
+  MuiPickersUtilsProvider,
+  KeyboardDateTimePicker,
```
Comment (hieudan225) [inline step-capstone/src/components/TravelObjectForms/AddEvent.js:12]: the only change in add[TravelObject] files is replacing DateTimePicker with KeyBoardDateTimePicker. Please ignore the add[TravelObject] files.

Comment (hieudan225) [reply step-capstone/src/components/TravelObjectForms/AddEvent.js:12]: Sorry for the confusion.

File: step-capstone/src/components/Firebase/firebase.js [hunk 2]
```diff
@@ -93,6 +93,7 @@ class Firebase {
 
     if (oldValue.destinations !== newValue.destinations) {
       await tripRef.update({ destinations: newValue.destinations });
+      await tripRef.update({ photoUrl: newValue.photoUrl})
```
Comment (emaohn) [inline step-capstone/src/components/Firebase/firebase.js:96]: nit: space after `newValue.photoUrl`

**Pull request merged**